### ISSUE TITLE
Visual C++ compiler warnings by bit shift by negative value

### ISSUE THREAD

**Issue by MinoruSekine** [MinoruSekine/libfixedpointnumber#94]
```
D:\Users\mino\work\libfixedpointnumber\include\fixedpointnumber_conversion-priv.h(127,1): warning C4293: '<<': シフト数が負の値であるか、大きすぎます。定義されていない動作です (ソース ファイルをコンパイルしています ..\..\..\..\sample\pi_by_leibniz.cc)
```

happens.

In resolving this issue, warning level for Windows should be more strict.

**Comment by MinoruSekine** [comment]
negative shift branch pointed out by this warning never be taken,
because it takes another branch of conditional operator if shift operand is negative.
But we need some tricky codes to shut up this warning by code modification. It is not reasonable.
So we consider to use `#pragma` to shut it up.

**Comment by MinoruSekine** [comment]
Resolved in #98 , and treat other warnings as errors in #99 .
